src/ray/core_worker/core_worker.cc

```diff
@@ -624,12 +654,13 @@ Status CoreWorker::AllocateReturnObjects(
     bool object_already_exists = false;
     std::shared_ptr<Buffer> data_buffer;
     if (data_sizes[i] > 0) {
-      if (!worker_context_.CurrentActorUseDirectCall()) {
+      if (worker_context_.CurrentActorUseDirectCall() &&
+          data_sizes[i] < RayConfig::instance().max_direct_call_object_size) {
```

Comment:
Can you add a TODO to unify this with the python version that checks whether we should put things in plasma?

Reply:
I thought about this more, and I don't think they really serve the same functionality. The Python check is done at the Python object level, and this is at the serialized object level. It's unclear whether either could substitute for the either entirely since they provide different benefits.